lib/filewatch/watch.rb

```diff
@@ -19,6 +19,8 @@ def initialize(opts={})
       @watching = []
       @exclude = []
       @files = Hash.new { |h, k| h[k] = Hash.new }
+      @unwatched = Hash.new
+      @lock = Mutex.new
```

Comment:
I'd provide more documentation/info on why this lock exists: because now there's an external operation - `unwatch` - called from tail.rb, that manipulates state, and can conflict with `each`, `discover` and `watch`